activerecord/lib/active_record/type/integer.rb

```diff
@@ -17,8 +17,7 @@ def type
       end
 
       def deserialize(value)
```

Comment:
Won't the behavior be the same if you just remove this method and let the superclass be called?